### ISSUE TITLE
Rearrange OG-USA's executable scripts

### ISSUE THREAD

**Issue by PeterDSteinberg** [open-source-economics/OG-USA#273]
I am working to get our conda packaging more smooth for OG-USA, Tax-Calculator and B-Tax.  OG-USA currently has the following files in the `./Python/` dir of the repository and these files are unfortunately not included in the conda package:
```
Python/execute.py
Python/execute_large.py
Python/execute_small.py
Python/postprocess.py
Python/run_many_ogusa_parallel.py
Python/run_ogusa.py
Python/run_ogusa_baseline.py
Python/run_ogusa_policy.py
Python/run_ogusa_serial.py
Python/run_simulation.py
Python/run_simulation_from_json.py
Python/run_small.py
Python/run_small_baseline.py
```

This limits the usefulness of an OG-USA conda package.  For example, the conda package currently cannot be used on dropq EC2 workers because dropq workers EC2 workers need some of the scripts not included in the conda package.

I propose the following changes to package structure that will include these files in the conda package and still make them easy to use for interactive development:

 * [ ] Create a `Python/scripts` directory with an `__init__.py` file
 * [ ] Move `.py` scripts that reside in the top level of `./Python` to `./Python/scripts`
 * [ ] Edit `setup.py` to create console entry points for each of the scripts being moved, as [shown in this example](https://calvinx.com/2012/09/09/python-packaging-define-an-entry-point-for-console-commands/)
 * [ ] Somewhat unrelated: Modify setup.py and MANIFEST.in so that users may run `python setup.py install` OR `python setup.py develop`. Currently `python setup.py develop` does not work as expected.

Basically the idea would be that users could do either one of the following pathways:

1) Install from a conda package
```
conda install -c ospc taxcalc ogusa btax
ogusa-execute
```
where `ogusa-execute` is the same thing as `python Python/execute.py` currently, and similar `ogusa-*` command line tools  exist for all scripts in `Python/scripts`

2) Install from source 
```
# Create an environment with taxcalc and other packages
cd OG-USA && python setup.py install
ogusa-execute
```

Thoughts @martinholmer @MattHJensen @jdebacker ?

**Comment by jdebacker** [comment]
@PeterDSteinberg I think this sounds ok, but let me pose a question and a comment.

Comment:  none of the `run_*.py` files are necessary to run OG-USA.  Rather they are various scripts to run specific policy proposals or to run OG-USA in a specific way  (e.g. in parallel).  They are nice to have in the repo so other users can references to modify for their needs, but they would not need to be in the conda package.

Question: Something I noticed with B-Tax has been that when I do install from `python setup.py install` and then I've changed some code while working in a branch, I need to redo `python setup.py install` before I run the model again to have those changes reflected in a run of B-Tax.  This slows down my development time because I've got to wait for the install with each change (and sometimes I run B-Tax before reinstalling and wonder why no change in results).  When I am running OG-USA (for which I've never done a `python setup.py install`), I can change a file, and then go directly to the command line to run with those changes in effect.    This is much faster.  My question is - is there a way to benefit from the ability to install the package through `python setup.py install`, but yet be able to develop code without having to reinstall after each change?  Should I just uninstall while making changes?  Or maybe my workflow is just completely silly - glad to take direction here as well.

cc @rickecon @kerkphil

**Comment by PeterDSteinberg** [comment]
@jdebacker - Regarding the `run_*.py` scripts: what you're saying makes sense for me.  If these `run_*.py` scripts are not really general purpose endpoints but rather example script templates one may edit, then we don't need to include them and this issue will just mean to move the `execute*.py` and `postprocess.py` to `Python/scripts`. 

For your question on B-Tax and wasted time with `python setup.py install`: B-Tax should be okay for you to do `python setup.py develop`.  If you do `python setup.py develop` then the Python packaging and import system will look for B-Tax in your local clone directory, thereby allowing you to edit the code and not re-install it.  One general note on using `python setup.py develop`: If you have already installed a package using `conda install` or `python setup.py *` and you then want to do `python setup.py develop` for that package, then it is advisable to run something like `conda remove btax;pip uninstall btax`.  That line, while not always required, ensures there is no confusion which `btax` you're using.  

As mentioned above, OG-USA has some issue that prevents `python setup.py develop` from working properly and the work on this issue will include fixing `python setup.py develop` for OG-USA.

**Comment by jdebacker** [comment]
@PeterDSteinberg - thank you for the answer.  I was not aware that one could do `python setup.py develop`.  Thats exactly the solution I was looking for.
